README.md

```diff
@@ -25,3 +25,56 @@ sudo chmod 666 /dev/ttyUSB0
 sudo chown $USER /dev/ttyUSB0
 sudo adduser $USER dialout
 ```
+# System Design
+
+## Modules
+Communication between the Hardware and Game Event Loop modules is done via message queue.
+Hardware events are placed in the queue by the Hardware module then the Game is ticked, this causes the Game to process the events, using the Player Engine to update the state of the game and enqueues Game messages into a queue for the Hardware system to process (resulting in animations to confirm actions were taken).
+The Game Event Loop uses functions in the player engine to update the state of the game.
+```mermaid
+graph
+    A[Hardware] --> B[(Game Message Queue)]
+    B --> C[Game Event Loop]
+    C --> D[(Hardware Message Queue)]
+    D --> A
+    C -..-> E[Player Engine]
+```
+
+### Persistent player state
+The persistent game state is stored in special hardware memory represented by a struct from the Player Engine.
+When the Hardware ticks the Game it sends the current player state.
+Then the Game calls functions from the Player Engine to transform the state based on the events it received from hardware.
+Once all messages are processed the persistent state is passed back to the hardware and the new state is stored in the special memory before sleep.
+
+### Hardware Module
+The hardware subsystem performs these steps on each wake-up:
+```mermaid
+graph TD
+    A(Wakeup) -->B[Scan Bluetooth]
+    B --> C[(Exposure Hw Msg)]
+    C --> D{Treatment?}
+    D -->|Yes| E[(Treatment Hw Msg)]
+    E --> F
+    D -->|No| F[Tick Game]
+    F --> Z{Game Msgs?}
+    Z -->|Yes| G[Animate Game Msgs]
+    G --> H(Sleep)
+    Z -->|No| H
+```
+
+### Game Module
+The Game event loop processes the messages placed in the queue from the hardware on each tick.
```

Comment:
So I guess this game messages thing is where we want to go?
Presumably the `game_update` would return/update this queue?
Which then mean there would be a hw engine to process them